### PR TITLE
Adjust the base64: env check to account for quotes

### PR DESCRIPTION
**Pull request by ericlbarnes**
This is in reference to #16186 which added support for prefixing env variables with `base64:` to have them automatically decoded. 

The original PR didn't account for the fact that variables with spaces need to be quoted in the .env file and this should support encoding those. For example: 

```
PRIVATE_KEY1=base64:mytest
PRIVATE_KEY2='base64:——BEGIN RSA PRIVATE KEY-----\n'
PRIVATE_KEY3="base64:——BEGIN RSA PRIVATE KEY-----\n"
```

### CODE REVIEW

File: src/Illuminate/Foundation/helpers.php [hunk 1]
```diff
@@ -456,8 +456,8 @@ function env($key, $default = null)
                 return;
         }
 
-        if (strpos($value, 'base64:') === 0) {
-            return base64_decode(substr($value, 7));
+        if (Str::startsWith($value, ['base64:', "'base64:", '"base64:'])) {
+            return base64_decode(trim($value, '"\''));
```
Comment (fernandobandeira) [inline src/Illuminate/Foundation/helpers.php]: Yes, the check looks better this way 😄

Comment (ericbarnes) [inline src/Illuminate/Foundation/helpers.php]: Thanks. I've pushed that substr addition. Also, I noticed we don't have any tests for this `env` helper but I'm unsure how to mock that getenv call so I didn't add any. :(

**Comment by vlakoff** [comment]
But... the variable is the base64-encoded value, not the plaintext.

**Comment by ericbarnes** [comment]
@vlakoff not sure what you mean?

**Comment by vlakoff** [comment]
Your example should actually be:

    PRIVATE_KEY1=base64:bXl0ZXN0
    PRIVATE_KEY2='base64:LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQ=='
    PRIVATE_KEY3="base64:LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQ=="

There can't be space, so there is no need for quoting.

However, if for consistency you want to support double quotes as in other variables, the following change could be made:

From:
```php
if (strpos($value, 'base64:') === 0) {
    return base64_decode(substr($value, 7));
}

if (strlen($value) > 1 && Str::startsWith($value, '"') && Str::endsWith($value, '"')) {
    return substr($value, 1, -1);
}

return $value;
```

To:
```php
if (strlen($value) > 1 && Str::startsWith($value, '"') && Str::endsWith($value, '"')) {
    $value = substr($value, 1, -1);
}

if (strpos($value, 'base64:') === 0) {
    return base64_decode(substr($value, 7));
}

return $value;
```

**Comment by ericbarnes** [comment]
Thanks for the clarification. I'm sorry I had this feature confused and without the spaces I don't think is needed at all. 

Can you think of any other examples besides a key file?

**Comment by vlakoff** [comment]
Actually RSA keys don't need base64 encode. Just remove comments and newlines. The key files are usually already base64 encoded…

I can't think of use cases for base64, but I bet people have found some.

About my above snippet, maybe we should just leave the code as it currently, i.e. just support base64: unquoted.

**Comment by ericbarnes** [comment]
Yeah I agree. I'll close this. I was working on writing up about this feature and couldn't think of an example and someone recommended key files as an example. That's what pushed me thinking about it from that angle. I appreciate all the clarification. 👍🏼

**Comment by fernandobandeira** [comment]
The only usecase that I can think is the one described in the original internals issue:

laravel/internals#266

> To encode passwords in the .env. Makes it a bit harder for someone to see a password over your shoulder.

**Pull request closed without merging**